### PR TITLE
Fixed a bug where Artifacts were missing some checksums after `podman…

### PR DESCRIPTION
**Pull request by ipanova**
… push`.

closes #7774
https://pulp.plan.io/issues/7774

### CODE REVIEW

**Comment by ipanova** [comment]
@dkliban @mdellweg  we need to provide a migration along this fix as well

**Comment by mdellweg** [comment]
you mean a data migration, agreed. If we need to make this backportable, we could provide it as an admin command. Adding the missing checksums is at least idempotent.

**Comment by mdellweg** [comment]
Can we create that artifact using its serializer, to avoid such problems in the future?

**Comment by ipanova** [comment]
> you mean a data migration, agreed. If we need to make this backportable, we could provide it as an admin command. Adding the missing checksums is at least idempotent.

There is some discussion on this ticket. Apparently we are going to add a django command along with the plugin providing the data migration https://pulp.plan.io/issues/7696
Once https://pulp.plan.io/issues/7696 and https://pulp.plan.io/issues/7561 will be addressed we would solve the issue of having missing checksums.
Speaking of the serializer, i don't know whether it would be able to cover all the cases mentioned in the discussion, at least we would need to extend it with the additional logic for the allowed_checksums. The pain point is that we cannot enforce plugins writers to  use the serialzer.

File: pulp_container/app/migrations/0010_fix_missing_blobs_checksums.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import hashlib
+from django.db import migrations, transaction
+
+
+CHUNK_SIZE = 1024 * 1024  # 1 Mb
+
+
+def fix_missing_blobs_checksums(apps, schema_editor):
+    with transaction.atomic():
```
Comment (mdellweg) [inline pulp_container/app/migrations/0010_fix_missing_blobs_checksums.py:11]: Why is this atomic?

**Review by mdellweg — COMMENTED** [review]
I realized, this will have a hard time to run, because migrations are called with pulpcore-admin and pulpcore-admin will already refuse to start if any artifact is improperly saved.

**Comment by ipanova** [comment]
> I realized, this will have a hard time to run, because migrations are called with pulpcore-admin and pulpcore-admin will already refuse to start if any artifact is improperly saved.

yes, i had same realization after i have written the code.. we will need to find a different solution to this

**Review by mdellweg — APPROVED** [review]

**Pull request merged**